src/CoreWCF.ConfigurationManager/tests/CustomBindingTests.cs

```diff
@@ -0,0 +1,765 @@
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.CloseTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.OpenTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.SendTimeout);
+                    Assert.Equal(expectedReceiveTimeout, actualBinding.ReceiveTimeout);
+                }
+            }
+        }
+
+        [Fact]
+        public void CustomBinding_WithDefaultSettings()
+        {
+            string expectedName = "customBinding";
+            TimeSpan expectedReceiveTimeout = TimeSpan.FromMinutes(10);
+            TimeSpan expectedDefaultTimeout = TimeSpan.FromMinutes(1);
+
+
+            string xml = $@"
+<configuration> 
+    <system.serviceModel>         
+        <bindings>         
+            <customBinding>
+                <binding name=""{expectedName}"">
+                </binding>
+            </customBinding>                             
+        </bindings>                             
+   </system.serviceModel>
+</configuration>";
+
+            using (var fs = TemporaryFileStream.Create(xml))
+            {
+                using (ServiceProvider provider = CreateProvider(fs.Name))
+                {
+                    IConfigurationHolder settingHolder = GetConfigurationHolder(provider);
+
+                    CustomBinding actualBinding = settingHolder.ResolveBinding(nameof(CustomBinding), expectedName) as CustomBinding;
+
+                    Assert.Equal(expectedName, actualBinding.Name);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.CloseTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.OpenTimeout);
+                    Assert.Equal(expectedDefaultTimeout, actualBinding.SendTimeout);
+                    Assert.Equal(expectedReceiveTimeout, actualBinding.ReceiveTimeout);
+                }
+            }
+        }
+
+        #endregion
+
+        #region Encoding
+
+        #region TextEncoding
+
+        [Fact]
+        public void CustomBinding_WithTextEncodingSettings()
+        {
+            string expectedName = "customBinding";
+            int expectedMaxDepth = 2147483647;
+            int expectedMaxReadPoolSize = 24;
+            int expectedMaxWritePoolSize = 8;
+            Encoding expectedEncoding = Encoding.Unicode;
+            MessageVersion expectedMessageVersion = MessageVersion.Soap11;
+
+
+            string xml = $@"
+<configuration> 
+    <system.serviceModel>         
+        <bindings>         
+            <customBinding>
+                <binding name=""{expectedName}"">
+                    <textMessageEncoding maxReadPoolSize=""{expectedMaxReadPoolSize}""
+                                         maxWritePoolSize=""{expectedMaxWritePoolSize}""
+                                         writeEncoding=""utf-16LE""
+                                         messageVersion=""Soap11"">
```

Comment:
You can use `{expectedEncoding.HeaderName}` to get the string used for Encoding.Unicode. I'm not sure if you want to use that or not though as what you currently have won't roundtrip. Calling `Encoding.GetEncoding("utf-16LE")` will return you the same as `Encoding.Unicode`, but the `HeaderName` is `utf-16` without the `LE` suffix as little endian is presumed. So if you were to regenerate the config from the binding, it would output `utf-16` which isn't what you have here. I would suggest using `{expectedEncoding.HeaderName}` but just know it's testing something slightly different than what you have here. If you want to test multiple encodings, you could use `[Theory]` instead of `[Fact]` and have the test run with different encodings. I'll leave it to you to decide, and leaving it as currently written is also fine.